app/Main.hs

```diff
@@ -244,52 +246,127 @@ findDelimiter (s:ss) =
   then s
   else findDelimiter ss
 
+data Command
+  = CmdParse
+  | CmdFeedback Bool
+  | CmdCheck
+  | CmdValid
+  | CmdShow
+  | CmdPending (Maybe Int)
+  | CmdStatus (Maybe Int)
+  | CmdSummary (Maybe Word)
+  | CmdExport (Maybe String)
+  | CmdExportResults
+  | CmdExportTable
+  | CmdExportHTML
+  | CmdExportMD
+  | CmdExportPdfMark
+  deriving (Show)
+
+data Args = Args
+  { argsCommand :: Command
+  , argsPaths :: [String]
+  }
+  deriving (Show)
+
+cmdFeedbackParser :: App.Parser Command
+cmdFeedbackParser =
+  CmdFeedback
+    <$> switch
+      ( long "with-points" )
+  <**> helper
+
+cmdWithDepthParser :: Read a => (Maybe a -> Command) ->  App.Parser Command
```

Comment:
I would rather use a `Word` or even `Word8` explicitly here, but the command implementations use `Int` for some reason, so let's fix that in a follow-up PR.

Reply:
> I would rather use a `Word` or even `Word8` explicitly here, but the command implementations use `Int` for some reason, so let's fix that in a follow-up PR.

That being the `pending` and `status` commands.

Reply:
Word? Is the depth a bitfield?

Reply:
> Word? Is the depth a bitfield?

It is a non-negative integer.

Reply:
Don't use unsigned integers for anything but bitfields. If you think there is a significant source of bugs to be solved by clarifying the type here, then use something like `Nat`, but consider whether you're just adding a bunch of noisy `fromIntegral` calls to the code, or whether you're really fixing a class of bugs.